src/rdb/rdb_raft.c

```diff
@@ -1855,9 +1866,11 @@ rdb_raft_append_apply_internal(struct rdb *db, msg_entry_t *mentry,
 			goto out;
 	}
 
+	ABT_mutex_lock(db->d_raft_mutex);
 	rdb_raft_save_state(db, &state);
 	rc = raft_recv_entry(db->d_raft, mentry, &mresponse);
 	rc = rdb_raft_check_state(db, &state, rc);
+	ABT_mutex_unlock(db->d_raft_mutex);
 	if (rc != 0) {
```

Comment:
(Not related to this "if"; just need a place to add a comment.) rdb_raft_{add,remove}_replica also need to hold d_raft_mutex.